Scripts/PlayerCamera.cs

```diff
@@ -0,0 +1,78 @@
+using Godot;
+
+public class PlayerCamera : Camera
+{
+    [Export]
+    private readonly float PlayerDistanceThreshold = 2f;
+    [Export]
+    private readonly float CursorDistanceThreshold = 1f;
+    [Export]
+    private readonly float OffsetZ = 25f;
+    [Export]
+    private readonly float Height = 50f;
+    [Export]
+    private readonly float LookRange = 5.0f;
+    [Export]
+    private readonly float FollowCursorSpeedFactor = 0.3f;
+    [Export]
+    private readonly float FollowPlayerSpeedFactor = 0.7f;
+    [Export]
+    private readonly float VelocityDamping = 0.9f;
+
+    private Player player;
+    private float TargetHeight;
+    private Vector3 followCursorVelocity;
+    private Vector3 followPlayerVelocity;
+    private Vector3 currentCursorOffset;
+
+    public override void _Ready()
+    {
+        player = GetNodeOrNull<Player>("../Player");
+        TargetHeight = Height;
+        followCursorVelocity = new Vector3();
+        followPlayerVelocity = new Vector3();
+        currentCursorOffset = new Vector3();
+    }
+
+    public override void _PhysicsProcess(float delta)
+    {
+        if (player.IsMovementLocked)
+            TargetHeight = Height * 0.75f;
```

Comment:
I feel that 0.75 is too much if the game is fast-paced and there is a lot of dashing.

Reply:
I think we should just change this out entirely